### ISSUE TITLE
Random "User session terminated" messages

### ISSUE THREAD

**Issue by jhfrontz** [tigase/beagle-im#53]
**Describe the bug**
Every once in a while, I will see "User session terminated" inserted into a DM conversation.  In this latest instance,  they were inserted after the last few messages (that varied from 1-3 minutes old)

**To Reproduce**
Not sure

**Expected behavior**
Some sort of indication of what is going on (does it meant that the server died or the other party's client died?  Were the messages actually transmitted?)

**Screenshots**
![session-term](https://user-images.githubusercontent.com/2057172/94166771-bb87b900-fe59-11ea-8a67-31c230dd5655.jpg)

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser N/A
 - Version 10.14.6

**Additional context**

**Comment by hantu85** [comment]
You receive those errors (those are message of type `error`) sent back to you by the remote server (recipients XMPP server).

This is not a bug. This is normal response of a client upon receiving a message of type `error`. If those messages would be lost, you would want to be notified, right?

The bug is on the remote server as it should not sent those messages of type `error` if those message were delivered to the end user's device or stored in the offline storage.

**Comment by jhfrontz** [comment]
What's confusing is that they seem to be inserted into already seen portions of the conversation (I mean, if the other party hadn't seen them, he wouldn't have replied what he did).  Is there some sort of persistent context for a message that allows a later status update?

**Comment by hantu85** [comment]
Messages (also those with type `error`) may contain timestamp details when they should be placed in a history [XEP-0203: Delayed Delivery](https://xmpp.org/extensions/xep-0203.html) and if that would be there, then Beagle will use that information to place the message in the history.

However, I suppose, that error is sent to you not when message with "text" is not delivered but when you receive a message with a response, then Beagle sends confirmation that it received that messages. I suppose that when this message is received by the remote server, the recipient is gone (disconnected) and remote server sends an error back.

**Comment by jhfrontz** [comment]
I'm not sure how much context is available when those errors come from the far end -- but if there is enough to say "failed to get acknowledgement of confirmation" (or whatever is going on) that would be somewhat more helpful in allowing a user to diagnose what is going on.

**Comment by totaleo** [comment]
Agreed that this message a bit terrible. I have received more that 10 same messages in a one minute.
Version 4.1
![](https://habrastorage.org/webt/jo/8_/om/jo8_ommfzu3azt8wpj3bppbmrpi.png)
